server/base/src/main/java/org/apache/accumulo/server/master/state/MetaDataStateStore.java

```diff
@@ -60,25 +69,22 @@ protected MetaDataStateStore(ClientContext context, CurrentState state, String t
   }
 
   @Override
-  public void setLocations(Collection<Assignment> assignments) throws DistributedStoreException {
-    BatchWriter writer = createBatchWriter();
-    try {
-      for (Assignment assignment : assignments) {
-        Mutation m = new Mutation(assignment.tablet.getMetadataEntry());
-        assignment.server.putLocation(m);
```

Comment:
If some of the methods in Assignment are no longer used it would be nice to remove them

Reply:
Yeah, I am still working on that. Ran into issues in some of the tests that use it

Reply:
I removed the set and clear LastLoc functions but I am having trouble removing the current and future location functions. The setFutureLocation function in MetaDataStateStore uses SuspendingTserver.java which takes in a Mutation and Idk if it makes sense to use tabletmutator and a mutation for the same function so I left it as is. If you have ideas on how I should proceed there then I will. All use cases are in MetaDataStateStore.java and in a couple IT's.

Reply:
Also running into an issue in MasterRepairsDualAssignmentIT. I am unable to get the ample object for TabletMutator. That IT is the only use case left of the putLocation in TserverInstance then it can also be deleted.

Reply:
@keith-turner

Reply:
Removing those is important, but don't let it impeded progress on this issue.  I was just suggesting to move any that became unused in the course of making changes for this issue.  I think we should open up follow on issues about removing any that are still in use after this change.

Reply:
Okay, that makes sense. Removing those shouldn't be difficult. I am just unsure on how to alter SuspendingTServer to get it to work with TabletMutator instead of mutations.

Reply:
Let me know when you would like me to take another look at this.

Reply:
Will do. I will hopefully make a commit today on some of the assignment changes mentioned below.